src/ordered_set.jl

```diff
@@ -6,6 +6,7 @@
 struct OrderedSet{T}  <: AbstractSet{T}
     dict::OrderedDict{T,Nothing}
 
+    OrderedSet{T}(dict::OrderedDict{T,Nothing}) where {T} = new{T}(dict)
```

Comment:
For consistency with Base, do you think it would be better to make this take a `KeySet`?
So it would be `OrderedSet(keys(myordereddict))`?
That way it can be used on any `OrderedDict`, not just one with no values.

We would basically like to mirror the base code except with the signature:
`OrderedSet{T}(s::KeySet{T, <:OrderedDict{T}}) where {T}`
https://github.com/JuliaLang/julia/blob/78ec99b585c13bad56668162434e4266b85cca7e/base/set.jl#L50-L56

The other issue that would solve is that the mutating the OrderedDict would not mutate the OrderedSet that was created from it.
Admittedly it does come at a cost of a `copy` but sequential copies like this are super fast.

Reply:
The inner constructor `_Set` is a bit inelegant, but using the method `OrderedSet{T}(s::KeySet{T, <:OrderedDict{T}}) where {T}` as an inner constructor solves the issue of mirroring `Base`. This has been added.